R/getShortTermChange.R

```diff
@@ -0,0 +1,146 @@
+#'   used in calculation. (Default: \code{"counts"})
+#' @param rarefy \code{Logical scalar}. Whether to rarefy counts.
+#' (Default: \code{FALSE})
+#' @param compositional \code{Logical scalar}. Whether to transform counts.
+#' (Default: \code{FALSE})
+#' @param depth \code{Integer scalar}. Specifies the depth used in rarefying. 
+#' (Default: \code{min(assay(x, assay.type)}))
+#' @param ... additional arguments.
+#' 
+#' 
+#' @return \code{dataframe} with \code{short term change} 
+#' calculations.
+#' 
+#' @details This approach is used by Wisnoski NI and colleagues
+#'          \url{https://github.com/nwisnoski/ul-seedbank}. Their approach is based on
+#'          the following calculation log(present abundance/past abundance).
+#'          Also a compositional version using relative abundance similar to
+#'          Brian WJi, Sheth R et al
+#'          \url{https://www.nature.com/articles/s41564-020-0685-1} can be used.
+#'          This approach is useful for identifying short term growth behaviors of taxa.
+#'          
+#' @name getShortTermChange
+#' 
+#' 
+#' @examples
+#' 
+#' # Load time series data
+#' data(minimalgut)
+#' tse <- minimalgut
+#' 
+#' short_time_labels <- c("74.5h", "173h", "438h", "434h", "390h")
+#' 
+#' # Subset samples by Time_lable and StudyIdentifier
+#' tse <- tse[, !(tse$Time_label %in% short_time_labels)]
+#' tse <- tse[, (tse$StudyIdentifier == "Bioreactor A")]
+#' 
+#' # Get short term change
+#' getShortTermChange(tse, rarefy = TRUE, time.col = "Time.hr")
+NULL
+
+#' @rdname getShortTermChange
+#' @export
+setGeneric("getShortTermChange", signature = c("x"),
+    function( x, assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), ...)
+        standardGeneric("getShortTermChange"))
+
+#' @rdname getShortTermChange
+#' @export
+#' @importFrom dplyr arrange as_tibble summarize "%>%"
+#' @importFrom ggplot2 ggplot aes
+#' @importFrom ggrepel geom_text_repel
+#' @importFrom mia rarefyAssay transformAssay
+#' @importFrom SummarizedExperiment colData
+setMethod("getShortTermChange", signature = c(x = "SummarizedExperiment"),
+    function(x, assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), ...){
+        ############################## Input check #############################
+        # Check validity of object
+        if(nrow(x) == 0L){
+            stop("No data available in `x` ('x' has nrow(x) == 0L.)",
+                 call. = FALSE)
+        }
+        # Check assay.type
+        .check_assay_present(assay.type, x)
+        if(!.is_a_bool(rarefy)){
+            stop("'rarefy' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if(!.is_a_bool(compositional)){
+            stop("'compositional' must be TRUE or FALSE.", call. = FALSE)
+        }
+        # Ensure that the provided depth is valid
+        if ( !is.null(depth) && depth > min(assay(x, assay.type), na.rm = TRUE) ) {
```

Comment:
You are using styles `if (something)` and `if(something)`. Good to be consistent